http/1.6/modules/ROOT/pages/http-request-ref.adoc

```diff
@@ -39,41 +95,107 @@ By default, Mule defines different limits for the HTTP request size, response si
+----
+
+The following example shows how to configure the *Default headers* field at the HTTP *Request* global configuration level in Studio:
+
+. Select the HTTP *Request* operation from your flow.
+. In *General*, select the *Headers* tab.
+. Click the plus sign (*+*) next to the *Connector configuration* field to configure a global element that can be used by all instances of HTTP *Request* in the app.
+. Click *Settings*.
+. Set *Default headers* to *Edit inline*.
+. Click the plus sign (*+*) to add a new header.
+. Set *Key* to `x-csrf-token`.
+. Set *Value* to `Fetch`.
+. Click *Finish*.
+
+image::http-request-6.png[Default headers configuration at HTTP Request global connection]
+
+In the *Configuration XML* editor, the `default-headers` configuration looks like this:
+
+[source,xml,linenums]
 ----
+<http:default-headers >
+    <http:default-header key="x-csrf-token" value="Fetch" />
+</http:default-headers>
+----
+
+The following example shows how to configure default headers with DataWeave expressions, allowing you to use dynamic values:
 
-With this configuration, those headers will be added to every outbound request, alongside any headers you specify in the actual request.
+. In the *Default headers* configuration window.
+. For *Value*, select the *fx* button to switch to literal mode.
+. Set *Value* to `#[vars.role]`.
 
-The default headers also accept expressions, allowing you to use dynamic values. For example:
+image::http-request-7.png[Default headers configuration at HTTP Request global connection]
+
+In the *Configuration XML* editor, the `default-headers` configuration looks like this:
 
 [source,xml,linenums]
 ----
 <http:request-config name="requestConfig">
-    <http:request-connection host="localhost" port="8081" streamResponse="true" responseBufferSize="1024"/>
     <http:default-headers >
         <http:default-header key="custom-role" value="#[vars.role]" />
     </http:default-headers>
 </http:request-config>
 ----
 
-This practice is helpful and convenient but has some important caveats to consider. Using expressions in a configuration element constitutes a dynamic configuration. The lifecycle of a dynamic configuration is not explained here, but in a nutshell, all expressions in the config are evaluated each time the connector is used and for each set of distinct values a new configuration instance is created and initialized.
+Using DataWeave expressions in a configuration element constitutes a dynamic configuration. Each time a Mule app uses an HTTP *Request* operation, all expressions in the configuration are evaluated and for each set of distinct values, a new configuration instance is created and initialized.
+
+For HTTP Connector, use default headers with small value expressions. Otherwise, every single evaluation returns a different value and creates several instances of the HTTP client. This behavior affects performance because of the extra work that takes to initialize a client and can be resource-consuming.
```

Comment:
```suggestion
For HTTP Connector, use default headers with small value expressions. Otherwise, every single evaluation returns a different value and creates several instances of the HTTP client, consuming valuable resources and negatively affecting performance.
```